action.yml

```diff
@@ -192,21 +192,28 @@ runs:
         fi
         nvm install
         nvm use
-        rm -rf client/dist
         if [[ $(npm -v) != $ADMIN_NPM_VERSION ]]; then
           npm install -g yarn;
         fi
         yarn install --network-concurrency 1
         if [[ $(cat package.json | jq -r '.scripts.build') != 'null' ]]; then
+          if [[ -d client/dist ]]; then
+            echo "Deleting client/dist directory"
+            rm -rf client/dist
+          fi
           echo "Running yarn build"
           yarn run build
-          echo "Running git diff"
-          git add client
-          GIT_DIFF=$(git diff --name-status --relative=client && git diff --cached --name-status --relative=client)
-          if [[ $GIT_DIFF != "" ]]; then
-            echo "git diff:"
-            echo $GIT_DIFF
-            exit 1
+          if ! [[ -d client ]]; then
+            echo "No client directory, not running git diff"
```

Comment:
In this case we should `exit 1` so we know we need to make a change (either don't run js build for this module or move the css/js to the client dir)

Reply:
I think it's probably valid to have no client dir in at least some cases - https://github.com/silverstripe/cwp-starter-theme

Reply:
I don't see why the css and js shouldn't be in the client dir for that package?

And if they do remain outside of `client/` for whatever reason, we should still be running the same CI tests against them so we need some way of identifying where they really are and running checks against that dir (in this case `src/`)

Reply:
Can't move them since they're a theme.  Sites will have things like `Requirements::css('themes/cwp-starter/dist/css/something.css');` - if we move the files to `client` then we break sites.

Reply:
Yeah good point. We'll have to run it against src in this case.
Not sure how we tell CI to do that though.

Reply:
Redone solution